packages/serverless/src/integrations/aws_lambda.ts

```diff
@@ -0,0 +1,235 @@
+import { EventProcessor, Hub, Integration, Scope } from '@sentry/types';
+import * as Sentry from '@sentry/node';
+
+/**
+ * NodeJS integration
+ *
+ * Provides a mechanism for NodeJS
+ * that raises an exception for handled, unhandled, timeout and similar times of error
+ * and captures the same in Sentry Dashboard
+ */
+export class AWSLambda implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'AWSLambda';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = AWSLambda.id;
+
+  /**
+   * Returns current hub
+   */
+  private getCurrentHub?: () => Hub;
+
+  /**
+   * context.
+   */
+  private awsContext: any;
+
+  /**
+   * timeout flag.
+   */
+  private timeoutWarning: boolean = false;
+
+  /**
+   * configured time in miliseconds.
+   */
+  private configuredTimeInMilliseconds: number;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(
+    options: { awslambda?: any; context?: any; timeoutWarning?: any; configuredTimeInMilliseconds?: number } = {},
+  ) {
+    this.awsContext = options.context;
+    this.timeoutWarning = options.timeoutWarning;
+    this.configuredTimeInMilliseconds = options.configuredTimeInMilliseconds as number;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    // lambda-bootstrap
+    const lambdaBootstrap = require.main;
+
+    if (!this.awsContext && !lambdaBootstrap) {
+      return;
+    }
+
+    const nodeJs: string = 'node';
+
+    /** capturing this context */
+    const setupOnceContext = this;
+
+    /** rapid runtime instance */
+    const rapidRuntime = lambdaBootstrap?.children[0].exports;
+
+    /** handler that is invoked in case of unhandled and handled exception */
+    const originalPostInvocationError = rapidRuntime.prototype.postInvocationError;
+
+    this.getCurrentHub = getCurrentHub;
+
+    const hub = this.getCurrentHub && this.getCurrentHub();
+
+    /** configured time */
+    this.configuredTimeInMilliseconds = this.awsContext.getRemainingTimeInMillis();
+
+    const timeoutWarningBuffer: number = 1500;
```

Comment:
Why is it constant 1500ms? It would be nice to have an explanation here.

Reply:
The constant is specific to timeout warning exception which we are sending to Sentry dashboard as an event.
We are sending timeout warning exception in case of timeout error and in parallel to function execution, so, we need to make sure that the timeout warning exception is sent before the function execution terminates, basically, it should be sent as late as possible but before the function execution terminates as it is a warning and not an actual timeout exception.
Now, after experimenting a little, it was observed that it takes roughly 1.349 seconds for the exception to get thrown and captured in Sentry dashboard after it is caught as an event.
Now, we turned that constant to 1.5 as an agreed upon value assuming some buffer time from the 1.349 seconds value.